sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -145,23 +152,17 @@ private static HttpClient CreateDefaultClient()
             };
         }
 
-        private static HttpMessageHandler CreateDefaultHandler()
+        private static HttpMessageHandler CreateDefaultHandler(HttpPipelineTransportOptions? options = null)
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))
             {
-                return new HttpClientHandler();
+                return ApplyOptionsToHandler(new HttpClientHandler(), options);
```

Comment:
If we don't call ApplyOptionsToHandler here we can drop all other `IsOSPlatform` checks